### DISCUSSION TITLE
cache problem

### DISCUSSION THREAD

**Discussion by Abdullah05-js** [TanStack/query#8344] [Q&A]
hello guys am having a problem when i go to another page and come back to my infinite post page the cache is gone 
![Ekran görüntüsü 2024-11-24 143742](https://github.com/user-attachments/assets/d2c41e07-1322-4110-87ff-4da1b87caa66)

**Comment by Abdullah05-js** [comment]
<li className='transition ease-in-out duration-1000 hover:translate-x-6'><Link to={"/home/profile"}><Button color="primary" variant="light" startContent={<Badge content={"5"}  color="success" variant="solid" placement="top-right" ><ProfileIcon/></Badge>} size='lg' className=" text-3xl   text-white    h-16">Profile</Button></Link></li>


this is the button thats navigate me to profile page when i go there and come back the posts key is gone (i can see it gone from the devtools)

**Reply by TkDodo** [reply]
https://tkdodo.eu/blog/react-query-fa-qs#2-the-queryclient-is-not-stable

**Reply by Abdullah05-js** [reply]
hello thanks for answering firstable i have the new QueryClient(); object outside the app 

![Ekran görüntüsü 2024-12-08 210505](https://github.com/user-attachments/assets/aeba9f69-6308-4050-ab1f-7735812e4614)

 my seneryo is like this i have profile page that  use useinfinitequery to get the user posts but when i go to another page the key that i put in the useinfinitequery is gone from the tanstack devtools 

![Ekran görüntüsü 2024-12-08 210626](https://github.com/user-attachments/assets/b4c19f8b-e9b7-4b63-b6b8-93c68023e649)
![Ekran görüntüsü 2024-12-08 210614](https://github.com/user-attachments/assets/71fe4d99-e063-4311-baea-289f715a456a)

the button that navigate me to profile page 

![Ekran görüntüsü 2024-12-08 210626](https://github.com/user-attachments/assets/8924af0a-d2f0-4c14-aaa3-061889e5c8b1)
